pinot-common/src/main/java/com/linkedin/pinot/common/query/context/TimerContext.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.common.query.context;
+
+import com.linkedin.pinot.common.metrics.ServerMetrics;
+import com.linkedin.pinot.common.metrics.ServerQueryPhase;
+import com.linkedin.pinot.common.request.BrokerRequest;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+public class TimerContext {
+
+  private final ServerMetrics serverMetrics;
+  private final Map<ServerQueryPhase, Timer> phaseTimers = new HashMap<>();
+  private final BrokerRequest brokerRequest;
+  private long queryArrivalTimeNanos;
+  
+  public class Timer implements AutoCloseable {
+    private final ServerQueryPhase queryPhase;
+    long startTime;
+    long endTime;
+    boolean isRunning;
+
+    Timer(ServerQueryPhase phase) {
+      this.queryPhase = phase;
+    }
+
+    void start() {
+      startTime = System.nanoTime();
+      isRunning = true;
+    }
+
+    void setStartTime(long startTime) {
```

Comment:
setStartTimeNs( long startTimeNs)
makes it more readable to have the unit in the method, thanks